src/main/frontend/src/app/app.service.ts

```diff
@@ -13,33 +13,75 @@ export type GitHubState = (typeof GitHubStates)[keyof typeof GitHubStates];
   providedIn: 'root',
 })
 export class AppService {
+  private readonly WITH_CREDENTIALS: string = 'withCredentials';
   private http = inject(HttpClient);
 
   /**
    * Performs a GET request to the given URL with optional query parameters.
    *
    * @template T - The expected response type.
    * @param url - The API endpoint URL.
-   * @param parameters - Optional query parameters as key-value pairs. Values must be primitive types.
+   * @param parameters - Optional query parameters as key-value pairs. Values can be primitive types or HTTP options.
    * @returns An Observable of type `T` containing the response data.
    */
-  public get<T>(url: string, parameters?: Record<string, string | number>): Observable<T> {
+  public get<T>(url: string, parameters?: Record<string, string | number | boolean>): Observable<T> {
     let httpParameters = new HttpParams();
+    const options: { params?: HttpParams; withCredentials?: boolean } = {};
 
     if (parameters) {
       for (const key of Object.keys(parameters)) {
         const value = parameters[key];
+
+        if (key === this.WITH_CREDENTIALS) {
+          options.withCredentials = value as boolean;
+          continue;
+        }
```

Comment:
Is this really needed?
You already have an http interceptor set up to add withCredentials.
This is also a bit confusing logic why do these custom get & post functions need to exist?